csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -97,7 +97,17 @@ public void TestSessionOptions()
                 opt.AppendExecutionProvider_CUDA(0);
 #endif
 #if USE_DML
+                // Explicitly set dll probe path so that the (potentially) stale system DirectML.dll 
+                // doesn't get loaded by the test process when it is eventually delay loaded by onnruntime.dll
+                // The managed tests binary path already contains the right DirectML.dll, so use that
+
+                var directml_dll_path = AppDomain.CurrentDomain.BaseDirectory;
+                SetDllDirectory(directml_dll_path);
                 opt.AppendExecutionProvider_DML(0);
+
+                // Restore the default dll search order
+                SetDllDirectory(null);
```

Comment:
Avoid polluting the entire process and restore the default search order after the right DirectML dll has been loaded (should be loaded when the dml EP is initialized in the AppendExecturionPeovider() call)